cocos/platform/CCFileUtils.cpp

```diff
@@ -1101,6 +1105,11 @@ void FileUtils::listFilesRecursivelyAsync(const std::string& dirPath, std::funct
     }, std::move(callback));
 }
 
+FileStream* FileUtils::openFileStream(const std::string& filePath, FileStream::Mode mode)
```

Comment:
1. Maybe add matched closeFileStream is better
2. Maybe openFileStream retrun nullptr when open failed, such as
```cpp
PosixFileStream fs;
if(fs.open(fullPath, mode)) return new PosixFileStream(std::move(fs));
return nullptr;
```

Reply:
Yes that is possible, I'll just have to add a default constructor to it so it will allow `PosixFileStream fs;` to work.

What do you think about renaming `FileUtils::openFileStream` to `FileUtils::createFileStream`? That way it won't confuse things with the `open` in the name, and we won't need to add a `closeFileStream()` method, since closing the stream should be as simple as "stream->close()".

The only type user could would know is FileStream.  No code outside openFileStream should ever create a concrete implementation of a FileStream (like PosixFileStream).

Reply:
It's may ok, my purpose is reduce new/delete when open file failed.

Reply:
Changes done.  I didn't change the `openFileStream()` name, since the name does represent the purpose of that method.

All usages of openFileStream now check for a nullptr on a failed open.

Reply:
ok, looks good to me

Reply:
Merge it

Reply:
Well, I have a question, does the openFileStream require fullPath?

Reply:
It depends on the implementation of FileStream.  In terms of the default PosixFileStream, the value passed in would need to be a full path, but if FileStream was a wrapper around a virtual file system, then it wouldn't need to be a full path. Does that make sense?

Reply:
> Merge it

I don't have permission on this project to merge.  Also, I haven't checked all code yet for any direct file access, so I'll post back here when I've reviewed the rest of the code for any remaining `fopen` and such.